code/modules/clothing/clothing.dm

```diff
@@ -438,12 +438,12 @@ SEE_PIXELS// if an object is located on an unlit area, but some of its pixels ar
 BLIND     // can't see anything
 */
 
-/proc/generate_female_clothing(index,t_color,icon,type)
+/proc/generate_female_clothing(t_color, icon, type, greyscale_colors)
 	var/icon/female_clothing_icon = icon("icon"=icon, "icon_state"=t_color)
 	var/icon/female_s = icon("icon"='icons/mob/clothing/under/masking_helpers.dmi', "icon_state"="[(type == FEMALE_UNIFORM_FULL) ? "female_full" : "female_top"]")
 	female_clothing_icon.Blend(female_s, ICON_MULTIPLY)
 	female_clothing_icon = fcopy_rsc(female_clothing_icon)
-	GLOB.female_clothing_icons[index] = female_clothing_icon
+	GLOB.female_clothing_icons[t_color + greyscale_colors] = female_clothing_icon
```

Comment:
When you're combining the two values, you should use some form of separator so we don't end up with some stupid duplicates issue in future

If t_color is "a#44ff20" and someone adds an icon state of a and a greyscale color of "#44ff20", they'll collide.

You can prevent this with something like `GLOB.female_clothing_icons["[t_color]-[greyscale_colors]"] = female_clothing_icon`

it's not an issue here, but it's just a good pattern